cluster/src/main/java/io/atomix/cluster/impl/DefaultClusterMembershipService.java

```diff
@@ -162,13 +163,19 @@ private void handleBroadcastMessage(byte[] message) {
   /**
    * Sends heartbeats to all peers.
    */
-  private CompletableFuture<Void> sendHeartbeats() {
+  private synchronized CompletableFuture<Void> sendHeartbeats() {
+    Member localMember = getLocalMember();
+    if (!localMember.metadata().equals(localMetadata)) {
```

Comment:
wouldn't double checked locking make more sense here?

Reply:
```java
  private CompletableFuture<Void> sendHeartbeats() {
    Member localMember = getLocalMember();
    if (!localMember.metadata().equals(localMetadata)) {
      sync(this) {
        if (!localMember.metadata().equals(localMetadata)) {
          localMetadata = localMember.metadata();
          post(new ClusterMembershipEvent(ClusterMembershipEvent.Type.MEMBER_UPDATED, localMember));
        }
      }
    }
```

Reply:
I was just being lazy :-P

Reply:
Less sync ftw.

Reply:
Now I have to fix it because you called me out...